demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -865,8 +890,8 @@ static void prvAzureDemoTask( void * pvParameters )
         /* Publish messages with QoS1, send and process Keep alive messages. */
         for( ; ; )
         {
-            ulScratchBufferLength = snprintf( ( char * ) ucScratchBuffer, sizeof( ucScratchBuffer ),
-                                              sampleazureiotMESSAGE, xDeviceCurrentTemperature );
+            prvCreateTelemetryPayload(ucScratchBuffer, sizeof(ucScratchBuffer), &ulScratchBufferLength);
```

Comment:
lets keep it simple with snprintf....To do that please add nano-spec, and uninitialize the printf/scanf.